packages/webdriverio/src/commands/element/moveTo.ts

```diff
@@ -23,26 +21,29 @@ type MoveToOptions = {
  */
 export async function moveTo (
     this: WebdriverIO.Element,
-    { xOffset, yOffset }: MoveToOptions = {}
+    { xOffset, yOffset }: MoveToOptions = {},
 ) {
     if (!this.isW3C) {
         return this.moveToElement(this.elementId, xOffset, yOffset)
     }
-    const isIntoView = async () => {
-        return await browser.execute(isElementInViewportScript, {
-            [ELEMENT_KEY]: this.elementId, // w3c compatible
-            ELEMENT: this.elementId // jsonwp compatible
-        } as any as HTMLElement)
-    }
     /**
      * W3C way of handle the mouse move actions
      */
     const browser = getBrowserObject(this)
-    await this.scrollIntoView({ block : 'nearest', inline: 'nearest', behavior: 'instant' })
-    if (!(await isIntoView())) {
-        await this.scrollIntoView({ block : 'center', inline: 'center', behavior: 'instant' })
+    const moveToNested = async ( xOffset?: number, yOffset?: number ) => {
+        return  await browser.action('pointer', { parameters: { pointerType: 'mouse' } })
+            .move({ origin: this, x: xOffset ? xOffset : 0, y: yOffset ? yOffset : 0 })
```

Comment:
Remove cast to integer and use click implementation

Reply:
What do you mean by cat to integer ?

Reply:
done